src/book/03-guides/04-general/01-git.mdx

```diff
@@ -0,0 +1,61 @@
+---
+section: Guides
+chapter: General
+title: Git
+description: Contributing to a NUbots project with git.
+slug: /guides/general/git
+---
+
+## Prerequisites
+
+This guide assumes you know a few basic shell commands, and the basics of git, such as adding, committing, and pushing to a repository.
+
+## Getting Code Into a Repository
+
+Let's suppose you have created a file which you would like to add to a NUbots repository called `MakeRobotStand.cpp`, which is part of your project to make the robot stand up.
+
+In the terminal, change directory to the location where the repository is cloned.
+
+### Select Branch
+
+If you are working within an existing project, then you need to find out which branch the work is on. To list the branches, use the command `git branch -r`. When you have found the branch you want to push to, swap to it using the following command:
+
+```sh
+git checkout <branch name>
+```
+
+Where `<branch name>` corresponds to the branch you found in the last step.
+
+If you are not working on an existing project, you need to make a branch to commit your changes to. The format for branch names is `surname/project_name`. So for this example it could be `citizen/stand_up`.
+
+### Making a Pull Request
+
+Once you have a set of commits on your branch which you think should be added to the main codebase, you can make a Pull Request (PR). PRs are where users ask the repository to pull their branch into the `master` branch.
+
+GitHub has documentation about PRs [here.](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request)
+
+Title your PR with a description of the changes that it will make. For this example we could use `Add stand-up script`. If you need to give more detail about what the code does, put that as a comment.
+
+To be approved, you should add reviewers to the pull request. If you have a project mentor, they should be a reviewer, as well as someone else who knows about that part of the codebase. Once the reviewers have approved the changes, the changes can be merged into the master branch.
+
+### Code Review
+
+The reviewers will give constructive feedback about your code, by making comments on code snippets from your changes. You can respond to these on GitHub. They can also directly suggest changes to specific sections. Additional commits you make to the branch will update the PR and will reset reviews of the changed sections.
+
+Once all of the comments are resolved, the reviewers have approved the PR, and the build checks have passed (see [_Formatting and Buildkite Checks_](#formatting-and-buildkite-checks)), you can `Squash and Merge`. Squashing removes detailed commit descriptions so that the `master` branch history is clean and easy to follow. Once the code has been squashed and merged, it is in the codebase and you have succeeded in contributing code to NUbots.
```

Comment:
Not code 😛 + wording

```suggestion
Once all of the comments are resolved, the reviewers have approved the PR, and the build checks have passed (see [_Formatting and Buildkite Checks_](#formatting-and-buildkite-checks)), you can **Squash and Merge**. Squashing combines every commit on the PR into a single commit so that the `master` branch history is clean and easy to follow. Once the code has been squashed and merged, it is in the codebase and you have succeeded in contributing code to NUbots.
```